File: 3-javascript/file.js
// file.js

//                      0        1         2         3          4
var colorsArray = [ "indigo", "black", "purple", "indigo", "chartreuse", "indigo", "indigo", "magenta", "chartreuse", "scarlet" ];


function countColors (color, theArray) {
  var counter;
  var colorOccurances = 0;

  for (counter = 0; counter < theArray.length; counter++) {
    var colorFromArray = theArray[counter];

    if (colorFromArray === color) {
      colorOccurances += 1;
    }
  }

  return colorOccurances;
}


console.log("Testing countColors function");


// p countColors("indigo", colors_array) == 4
console.log( countColors("indigo", colorsArray) === 4 );

// p countColors("chartreuse", colors_array) == 2
console.log( countColors("chartreuse", colorsArray) === 2 );

// p countColors("black", colors_array) == 1
console.log( countColors("black", colorsArray) === 1 );

// p countColors("pink", colors_array) == 0
console.log( countColors("pink", colorsArray) === 0 );

File: 3-javascript/file.rb
# file.rb

#                   0        1         2         3          4
colors_array = [ "indigo", "black", "purple", "indigo", "chartreuse", "indigo", "indigo", "magenta", "chartreuse", "scarlet" ]


def count_colors(color, the_array)
  color_occurances = 0

  the_array.each do |color_from_array|
    if color_from_array == color
      color_occurances += 1
    end
  end

  color_occurances
end


puts "Testing count_colors method"


p count_colors("indigo", colors_array) == 4

p count_colors("chartreuse", colors_array) == 2

p count_colors("black", colors_array) == 1

p count_colors("pink", colors_array) == 0



upper_case_colors = colors_array.map do |color_from_array|
  color_from_array.upcase
end

p upper_case_colors
